ibis/expr/api.py

```diff
@@ -648,6 +648,33 @@ def random() -> ir.FloatingScalar:
 
     Similar to [](`random.random`) in the Python standard library.
 
+    ::: {.callout-note}
+    `ibis.random()` will generate column(s) of distinct random numbers even if
+    the same instance of `ibis.random()` is re-used.
+
+    When Ibis compiles an expression to SQL, each place where `random` is used
+    will render as a separate call to the given backend's random number
+    generator.
+
+    ```python
+    >>> from ibis.interactive import *
+    >>> t = ibis.memtable({"a": range(5)})
+    >>> r_a = ibis.random()
+    >>> t.mutate(random_1=r_a, random_2=r_a)
```

Comment:
Probably want to skip here due to indeterminacy.

```suggestion
    >>> t.mutate(random_1=r_a, random_2=r_a)  # doctest: +SKIP
```